### ISSUE TITLE
brew install --HEAD neovim fails: Undefined symbols for architecture x86_64

### ISSUE THREAD

**Issue by cristian-heredia** [neovim/neovim#14923]
### Steps to reproduce using

```shell
❯ brew install --HEAD neovim
==> Downloading https://github.com/libmpack/libmpack-lua/releases/download/1.0.8/libmpack-lua-1.0.8.tar.gz
Already downloaded: /Users/cristian/Library/Caches/Homebrew/downloads/4b91349796a66e38e93acd77e6ca14212c869090429fd88271def2be5293d83f--libmpack-lua-1.0.8.tar.gz
==> Downloading https://luarocks.org/manifests/gvvaughan/lpeg-1.0.2-1.src.rock
Already downloaded: /Users/cristian/Library/Caches/Homebrew/downloads/0bc17975f0cf3981083172aa7da329b874e5780fa81494f5dd07d0665fcec23b--lpeg-1.0.2-1.src.rock
==> Cloning https://github.com/neovim/neovim.git
Updating /Users/cristian/Library/Caches/Homebrew/neovim--git
==> Checking out branch master
Already on 'master'
Your branch is up to date with 'origin/master'.
HEAD is now at 3552916cb fix(doc): remove reference to vim.lsp.callbacks (#14576)
==> luarocks make --lua-dir=/usr/local/opt/luajit-openresty --tree=/private/tmp/neovim-20210628-38550-6im1ya/deps-build
==> luarocks make --lua-dir=/usr/local/opt/luajit-openresty --tree=/private/tmp/neovim-20210628-38550-6im1ya/deps-build
==> cmake ..
==> make install
Last 15 lines from /Users/cristian/Library/Logs/Homebrew/neovim/04.make:
cd /tmp/neovim-20210628-38550-6im1ya/build/src/nvim && /usr/local/Homebrew/Library/Homebrew/shims/mac/super/clang -DINCLUDE_GENERATED_DECLARATIONS -DMIN_LOG_LEVEL=3 -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -D_GNU_SOURCE -Dnvim_EXPORTS -I/tmp/neovim-20210628-38550-6im1ya/build/config -I/tmp/neovim-20210628-38550-6im1ya/src -I/tmp/neovim-20210628-38550-6im1ya/build/src/nvim/auto -I/tmp/neovim-20210628-38550-6im1ya/build/include -I/usr/local/opt/luajit-openresty/include/luajit-2.1 -isystem /usr/local/opt/gettext/include -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -O2 -DNDEBUG -flto=thin -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wmissing-prototypes -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fno-common -fdiagnostics-color=auto  -Wno-conversion -MD -MT src/nvim/CMakeFiles/nvim.dir/xdiff/xprepare.c.o -MF CMakeFiles/nvim.dir/xdiff/xprepare.c.o.d -o CMakeFiles/nvim.dir/xdiff/xprepare.c.o -c /tmp/neovim-20210628-38550-6im1ya/src/nvim/xdiff/xprepare.c
[100%] Building C object src/nvim/CMakeFiles/nvim.dir/xdiff/xutils.c.o
cd /tmp/neovim-20210628-38550-6im1ya/build/src/nvim && /usr/local/Homebrew/Library/Homebrew/shims/mac/super/clang -DINCLUDE_GENERATED_DECLARATIONS -DMIN_LOG_LEVEL=3 -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -D_GNU_SOURCE -Dnvim_EXPORTS -I/tmp/neovim-20210628-38550-6im1ya/build/config -I/tmp/neovim-20210628-38550-6im1ya/src -I/tmp/neovim-20210628-38550-6im1ya/build/src/nvim/auto -I/tmp/neovim-20210628-38550-6im1ya/build/include -I/usr/local/opt/luajit-openresty/include/luajit-2.1 -isystem /usr/local/opt/gettext/include -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -O2 -DNDEBUG -flto=thin -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wmissing-prototypes -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fno-common -fdiagnostics-color=auto  -Wno-conversion -MD -MT src/nvim/CMakeFiles/nvim.dir/xdiff/xutils.c.o -MF CMakeFiles/nvim.dir/xdiff/xutils.c.o.d -o CMakeFiles/nvim.dir/xdiff/xutils.c.o -c /tmp/neovim-20210628-38550-6im1ya/src/nvim/xdiff/xutils.c
[100%] Linking C executable ../../bin/nvim
cd /tmp/neovim-20210628-38550-6im1ya/build/src/nvim && /usr/local/Cellar/cmake/3.20.5/bin/cmake -E cmake_link_script CMakeFiles/nvim.dir/link.txt --verbose=1
/usr/local/Homebrew/Library/Homebrew/shims/mac/super/clang  -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -O2 -DNDEBUG -flto=thin -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -Wl,-search_paths_first -Wl,-headerpad_max_install_names   -framework CoreFoundation CMakeFiles/nvim.dir/auto/msgpack_lua_c_bindings.generated.c.o CMakeFiles/nvim.dir/__/__/config/auto/pathdef.c.o CMakeFiles/nvim.dir/api/buffer.c.o CMakeFiles/nvim.dir/api/deprecated.c.o CMakeFiles/nvim.dir/api/private/dispatch.c.o CMakeFiles/nvim.dir/api/private/handle.c.o CMakeFiles/nvim.dir/api/private/helpers.c.o CMakeFiles/nvim.dir/api/tabpage.c.o CMakeFiles/nvim.dir/api/ui.c.o CMakeFiles/nvim.dir/api/vim.c.o CMakeFiles/nvim.dir/api/window.c.o CMakeFiles/nvim.dir/arabic.c.o CMakeFiles/nvim.dir/aucmd.c.o CMakeFiles/nvim.dir/autocmd.c.o CMakeFiles/nvim.dir/buffer.c.o CMakeFiles/nvim.dir/buffer_updates.c.o CMakeFiles/nvim.dir/change.c.o CMakeFiles/nvim.dir/channel.c.o CMakeFiles/nvim.dir/charset.c.o CMakeFiles/nvim.dir/context.c.o CMakeFiles/nvim.dir/cursor.c.o CMakeFiles/nvim.dir/cursor_shape.c.o CMakeFiles/nvim.dir/decoration.c.o CMakeFiles/nvim.dir/diff.c.o CMakeFiles/nvim.dir/digraph.c.o CMakeFiles/nvim.dir/edit.c.o CMakeFiles/nvim.dir/eval.c.o CMakeFiles/nvim.dir/eval/decode.c.o CMakeFiles/nvim.dir/eval/encode.c.o CMakeFiles/nvim.dir/eval/executor.c.o CMakeFiles/nvim.dir/eval/funcs.c.o CMakeFiles/nvim.dir/eval/gc.c.o CMakeFiles/nvim.dir/eval/typval.c.o CMakeFiles/nvim.dir/eval/userfunc.c.o CMakeFiles/nvim.dir/event/libuv_process.c.o CMakeFiles/nvim.dir/event/loop.c.o CMakeFiles/nvim.dir/event/multiqueue.c.o CMakeFiles/nvim.dir/event/process.c.o CMakeFiles/nvim.dir/event/rstream.c.o CMakeFiles/nvim.dir/event/signal.c.o CMakeFiles/nvim.dir/event/socket.c.o CMakeFiles/nvim.dir/event/stream.c.o CMakeFiles/nvim.dir/event/time.c.o CMakeFiles/nvim.dir/event/wstream.c.o CMakeFiles/nvim.dir/ex_cmds.c.o CMakeFiles/nvim.dir/ex_cmds2.c.o CMakeFiles/nvim.dir/ex_docmd.c.o CMakeFiles/nvim.dir/ex_eval.c.o CMakeFiles/nvim.dir/ex_getln.c.o CMakeFiles/nvim.dir/ex_session.c.o CMakeFiles/nvim.dir/extmark.c.o CMakeFiles/nvim.dir/file_search.c.o CMakeFiles/nvim.dir/fileio.c.o CMakeFiles/nvim.dir/fold.c.o CMakeFiles/nvim.dir/garray.c.o CMakeFiles/nvim.dir/getchar.c.o CMakeFiles/nvim.dir/hardcopy.c.o CMakeFiles/nvim.dir/hashtab.c.o CMakeFiles/nvim.dir/highlight.c.o CMakeFiles/nvim.dir/if_cscope.c.o CMakeFiles/nvim.dir/indent.c.o CMakeFiles/nvim.dir/indent_c.c.o CMakeFiles/nvim.dir/keymap.c.o CMakeFiles/nvim.dir/log.c.o CMakeFiles/nvim.dir/lua/converter.c.o CMakeFiles/nvim.dir/lua/executor.c.o CMakeFiles/nvim.dir/lua/treesitter.c.o CMakeFiles/nvim.dir/main.c.o CMakeFiles/nvim.dir/map.c.o CMakeFiles/nvim.dir/mark.c.o CMakeFiles/nvim.dir/marktree.c.o CMakeFiles/nvim.dir/math.c.o CMakeFiles/nvim.dir/mbyte.c.o CMakeFiles/nvim.dir/memfile.c.o CMakeFiles/nvim.dir/memline.c.o CMakeFiles/nvim.dir/memory.c.o CMakeFiles/nvim.dir/menu.c.o CMakeFiles/nvim.dir/message.c.o CMakeFiles/nvim.dir/misc1.c.o CMakeFiles/nvim.dir/mouse.c.o CMakeFiles/nvim.dir/move.c.o CMakeFiles/nvim.dir/msgpack_rpc/channel.c.o CMakeFiles/nvim.dir/msgpack_rpc/helpers.c.o CMakeFiles/nvim.dir/msgpack_rpc/server.c.o CMakeFiles/nvim.dir/normal.c.o CMakeFiles/nvim.dir/ops.c.o CMakeFiles/nvim.dir/option.c.o CMakeFiles/nvim.dir/os/dl.c.o CMakeFiles/nvim.dir/os/env.c.o CMakeFiles/nvim.dir/os/fileio.c.o CMakeFiles/nvim.dir/os/fs.c.o CMakeFiles/nvim.dir/os/input.c.o CMakeFiles/nvim.dir/os/lang.c.o CMakeFiles/nvim.dir/os/mem.c.o CMakeFiles/nvim.dir/os/process.c.o CMakeFiles/nvim.dir/os/pty_process_unix.c.o CMakeFiles/nvim.dir/os/shell.c.o CMakeFiles/nvim.dir/os/signal.c.o CMakeFiles/nvim.dir/os/stdpaths.c.o CMakeFiles/nvim.dir/os/time.c.o CMakeFiles/nvim.dir/os/tty.c.o CMakeFiles/nvim.dir/os/users.c.o CMakeFiles/nvim.dir/os_unix.c.o CMakeFiles/nvim.dir/path.c.o CMakeFiles/nvim.dir/popupmnu.c.o CMakeFiles/nvim.dir/profile.c.o CMakeFiles/nvim.dir/quickfix.c.o CMakeFiles/nvim.dir/rbuffer.c.o CMakeFiles/nvim.dir/regexp.c.o CMakeFiles/nvim.dir/runtime.c.o CMakeFiles/nvim.dir/screen.c.o CMakeFiles/nvim.dir/search.c.o CMakeFiles/nvim.dir/sha256.c.o CMakeFiles/nvim.dir/shada.c.o CMakeFiles/nvim.dir/sign.c.o CMakeFiles/nvim.dir/spell.c.o CMakeFiles/nvim.dir/spellfile.c.o CMakeFiles/nvim.dir/state.c.o CMakeFiles/nvim.dir/strings.c.o CMakeFiles/nvim.dir/syntax.c.o CMakeFiles/nvim.dir/tag.c.o CMakeFiles/nvim.dir/terminal.c.o CMakeFiles/nvim.dir/tui/input.c.o CMakeFiles/nvim.dir/tui/terminfo.c.o CMakeFiles/nvim.dir/tui/tui.c.o CMakeFiles/nvim.dir/ugrid.c.o CMakeFiles/nvim.dir/ui.c.o CMakeFiles/nvim.dir/ui_bridge.c.o CMakeFiles/nvim.dir/ui_compositor.c.o CMakeFiles/nvim.dir/undo.c.o CMakeFiles/nvim.dir/version.c.o CMakeFiles/nvim.dir/viml/parser/expressions.c.o CMakeFiles/nvim.dir/viml/parser/parser.c.o CMakeFiles/nvim.dir/window.c.o CMakeFiles/nvim.dir/xdiff/xdiffi.c.o CMakeFiles/nvim.dir/xdiff/xemit.c.o CMakeFiles/nvim.dir/xdiff/xhistogram.c.o CMakeFiles/nvim.dir/xdiff/xpatience.c.o CMakeFiles/nvim.dir/xdiff/xprepare.c.o CMakeFiles/nvim.dir/xdiff/xutils.c.o -o ../../bin/nvim  -fstack-protector-strong /usr/local/opt/gettext/lib/libintl.dylib /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/lib/libiconv.tbd /usr/local/lib/libluv.dylib /usr/local/lib/libuv.dylib -ldl -ldl /usr/local/lib/libmsgpackc.dylib /usr/local/lib/libvterm.dylib /usr/local/lib/libtermkey.dylib /usr/local/lib/libunibilium.dylib /usr/local/lib/libtree-sitter.dylib -lm -lutil /usr/local/opt/luajit-openresty/lib/libluajit-5.1.dylib
undef: _ts_query_cursor_set_match_limit
Undefined symbols for architecture x86_64:
  "_ts_query_cursor_set_match_limit", referenced from:
      _node_rawquery in lto.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [bin/nvim] Error 1
make[1]: *** [src/nvim/CMakeFiles/nvim.dir/all] Error 2
make: *** [all] Error 2

READ THIS: https://docs.brew.sh/Troubleshooting

Please create pull requests instead of asking for help on Homebrew's GitHub,
Twitter or any other official channels.
```

**Comment by stsewd** [comment]
Hi, as mentioned at https://github.com/neovim/neovim/pull/14915#issuecomment-869802740, this is homebrew issue.

**Comment by rickalex21** [comment]
Same problem here.

**Comment by xuliefeng** [comment]
+1

**Comment by iwfan** [comment]
![image](https://user-images.githubusercontent.com/28853200/123731374-97ff0c80-d8ca-11eb-9e34-34e76a98144c.png)

Same issue.

**Comment by clason** [comment]
Nothing we can do about it, please report it at https://github.com/Homebrew/homebrew-core/issues (although the homebrew maintainers are aware of it).

TL;DR: Homebrew (like other distros) replace the version of tree-sitter we specify in our build scripts for embedding by an earlier version to link against -- which doesn't have a function we now (have to) use.

Either `brew install tree-sitter --HEAD` as well or simply build neovim from this repo.
